Makefile

```diff
@@ -127,8 +149,16 @@ $(WEAVE_EXPORT): $(IMAGES_UPTODATE)
 $(DOCKER_DISTRIB):
 	curl -o $(DOCKER_DISTRIB) $(DOCKER_DISTRIB_URL)
 
-tests: tools/.git
-	tools/test
+ifeq ($(BUILD_IN_CONTAINER),true)
+tests: $(BUILD_UPTODATE) $(VENDOR_UPTODATE) tools/.git
+	$(SUDO) docker run $(RM) $(RUN_FLAGS) \
+		-v $(shell pwd):/go/src/github.com/weaveworks/weave \
```

Comment:
why is this not also mounting `.pkg`? Won't it end up building all libs from scratch as a result?